bigframes/core/compile/ibis_types.py

```diff
@@ -402,7 +404,12 @@ def literal_to_ibis_scalar(
             return bigframes_vendored.ibis.null()
 
     scalar_expr = bigframes_vendored.ibis.literal(literal)
-    if ibis_dtype:
+    if isinstance(literal, datetime.timedelta):
```

Comment:
Should we allow pandas, numpy, and pyarrow scalars here, too?

```suggestion
    if isinstance(literal, (datetime.timedelta, numpy.timedelta64, pandas.Timedelta, pyarrow.DurationScalar)):
```

Reply:
Sure! Pandas timedelta is a subclass of datetime.timedelta, but I can make it more explicit.

Added support for numpy.

I eventually decided now to implement pyarrow duration because:
* It causes more unrelated failures than I expected in Python 3.9 and 3.10 envs for some reasons.
* Pandas does not support adding pyarrow duration literals

Maybe we can add support for pyarrow in the future, but at this moment it may not be worth it